### PR TITLE
fix: unordered mode too many labels issue.

### PR DESCRIPTION
**Pull request by Genesis929**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: bigframes/session/executor.py [hunk 1]
```diff
@@ -379,11 +379,12 @@ def _run_execute_query(
...
                 bigframes.options.compute.maximum_bytes_billed
             )
-        # Note: add_labels is global scope which may have unexpected effects
-        bq_io.add_labels(job_config, api_name=api_name)
 
         if not self.strictly_ordered:
             job_config.labels["bigframes-mode"] = "unordered"
+
+        # Note: add_labels is global scope which may have unexpected effects
+        bq_io.add_labels(job_config, api_name=api_name)
```
Comment (shobsi) [inline bigframes/session/executor.py:387]: Highly recommend adding a test using the customer's use case

Comment (Genesis929) [reply bigframes/session/executor.py:387]: Added

File: tests/system/small/test_series.py [hunk 2]
```diff
@@ -2766,22 +2766,17 @@ def test_series_case_when(scalars_dfs_maybe_ordered):
     bf_series = scalars_df["int64_col"]
     pd_series = scalars_pandas_df["int64_col"]
 
-    # TODO(tswast): pandas case_when appears to assume True when a value is
```
Comment (shobsi) [inline tests/system/small/test_series.py:2769]: I would keep this comment

Comment (Genesis929) [reply tests/system/small/test_series.py:2769]: Updated to have all cases in the original test. Keep only one test can be overall faster I think.

**Review by shobsi — APPROVED** [review]

**Pull request merged**